CefSharp/Internals/JavascriptObjectRepository.cs

```diff
@@ -211,15 +213,31 @@ private void AnalyseObjectForBinding(JavascriptObject obj, bool analyseMethods,
                     var jsObject = CreateJavascriptObject(camelCaseJavascriptNames);
                     jsObject.Name = propertyInfo.Name;
                     jsObject.JavascriptName = GetJavascriptName(propertyInfo.Name, camelCaseJavascriptNames);
-                    jsObject.Value = jsProperty.GetValue(obj.Value);
+                    jsObject.LateBinding = () =>
+                    {
+                        if (!propertyInfo.PropertyType.IsArray)
+                        {
+                            jsObject.Value = jsProperty.GetValue(obj.Value);
+                            AnalyseObjectForBinding(jsObject, camelCaseJavascriptNames);
+                        }
+                        else
+                        {
+                            Array array = jsProperty.GetValue(obj.Value) as Array;
```

Comment:
Are you sure you want to do that? Type casting is a lot more expensive than just checking the flag. You did remind me that I need to check if array is null, else array.Length will throw an exception.